workflows/general/fastq-to-ubam.wdl

```diff
@@ -0,0 +1,146 @@
+                FO: "Flow order. The array of nucleotide bases that correspond to the nucleotides used for each flow of each read. Multi-base flows are encoded in IUPAC format, and non-nucleotide flows by various other characters. Format: /\\*|[ACMGRSVTWYHKDBN]+/",
+                KS: "The array of nucleotide bases that correspond to the key sequence of each read.",
+                LB: "Library.",
+                PG: "Programs used for processing the read group.",
+                PI: "Predicted median insert size, rounded to the nearest integer.",
+                PL: "Platform/technology used to produce the reads. Valid values: CAPILLARY, DNBSEQ (MGI/BGI), ELEMENT, HELICOS, ILLUMINA, IONTORRENT, LS454, ONT (Oxford Nanopore), PACBIO (Pacific Biosciences), SINGULAR, SOLID, and ULTIMA. This field should be omitted when the technology is not in this list (though the PM field may still be present in this case) or is unknown.",
+                PM: "Platform model. Free-form text providing further details of the platform/technology used.",
+                PU: "Platform unit (e.g., flowcell-barcode.lane for Illumina or slide for SOLiD). Unique identifier.",
+                SM: "Sample. Use pool name where a pool is being sequenced.",
+            },
+        }
+        prefix: "Prefix for the BAM file. The extension `.bam` will be added."
+        validate_input: "Ensure input BAM is well-formed before beginning harmonization?"
+        use_all_cores: "Use all cores? Recommended for cloud environments."
+    }
+
+    input {
+        Array[File] read_one_fastqs_gz
+        Array[File] read_two_fastqs_gz
+        Array[ReadGroup] read_groups
+        String prefix = basename(read_one_fastqs_gz[0], ".fastq.gz")
+        Boolean validate_input = true
+        Boolean use_all_cores = false
+    }
+
+    call parse_input { input:
+        read_one_fastqs_gz,
+        read_two_fastqs_gz,
+        read_groups,
+    }
+
+    scatter (read_group in read_groups) {
+        call read_group.validate_read_group as validate_readgroups { input:
+            read_group,
+            required_fields = ["ID", "SM", "LB", "CN", "DT"]
+        }
+    }
+
+    call ubam_core.fastq_to_ubam_core after parse_input after validate_readgroups { input:
+        read_one_fastqs_gz,
+        read_two_fastqs_gz,
+        read_groups,
+        prefix,
+        validate_input,
+        use_all_cores,
+    }
+
+    output {
+        File unaligned_bam = fastq_to_ubam_core.unaligned_bam
+    }
+}
+
+task parse_input {
+    meta {
+        description: "Parses and validates the `fastq-to-ubam` workflow's provided inputs"
+        outputs: {
+            check: "Dummy output to indicate success and to enable call-caching"
+        }
+    }
+
+    parameter_meta {
+        read_one_fastqs_gz: "Array of gzipped FASTQ files with 1st reads in pair"
+        read_two_fastqs_gz: "Array of gzipped FASTQ files with 2nd reads in pair"
+        read_groups: {
```

Comment:
I'd prefer saying refer to `../../data_structures/read_group.wdl` for more information than repeating it all here. Or future us will update `read_group.wdl` and forget to update these docs.

Reply:
Looks like you updated this in one spot but not the below spot.